cpu/cc2538/lpm.c

```diff
@@ -92,6 +92,16 @@ rtimer_clock_t lpm_stats[3];
  */
 static rtimer_clock_t sleep_enter_time;
 
+/* CC2538 silicon errata:
+   Possible Incorrect Value of Clock Dividers after PM2 and PM3 */
+#if SYS_CTRL_SYS_DIV == SYS_CTRL_CLOCK_CTRL_SYS_DIV_32MHZ || \
+  SYS_CTRL_IO_DIV == SYS_CTRL_CLOCK_CTRL_IO_DIV_32MHZ
+/* We use this variable in order to flag if we have adjusted either
+   one or both clock divisors upon entering PM2 or PM3. As everything
+   is done at compile time we can use defines to do the remaining stuff */
+static uint8_t reset_div;
+#endif
```

Comment:
No, you don't need such a flag. Just restore the dividers unconditionally. If they are already 0 (32 MHz), this won't change anything. And anyway, you have to make `select_32_mhz_xosc()` symmetrical to `select_16_mhz_rcosc()`because of the latter has to be changed because the dividers are not allowed to select clock frequencies higher than their source clocks (see UG), so you can't keep the system and I/O clock dividers set for 32 MHz when switching to the 16-MHz RC oscillator. See my code in my following comments on line 250 of `lpm.c`.